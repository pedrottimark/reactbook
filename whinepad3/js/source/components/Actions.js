/* @flow */

import React from 'react';

import type { Verb } from '../reducers/dialogue';

type Props = {
  onAction: (verb: Verb) => void,
};

const Actions = (props: Props) =>
  <div className="Actions">
    <span
      tabIndex="0"
      className="ActionsInfo"
      title="More info"
      onClick={props.onAction.bind(null, 'display')}>&#8505;</span>
    <span
      tabIndex="0"
      className="ActionsEdit"
      title="Edit"
      onClick={props.onAction.bind(null, 'update')}>&#10000;</span>
    <span
      tabIndex="0"
      className="ActionsDelete"
      title="Delete"
      onClick={props.onAction.bind(null, 'delete')}>x</span>
  </div>

Actions.defaultProps = {
  onAction: () => {},
};

export default Actions
