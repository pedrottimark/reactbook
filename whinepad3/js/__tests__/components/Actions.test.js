import React from 'react';
import TestUtils from 'react-addons-test-utils';
import renderer from 'react-test-renderer';

import Actions from '../../source/components/Actions';

// For TestUtils because Actions is a functional component.
class Wrap extends React.Component {
  render() {
    return <div>{this.props.children}</div>;
  }
}

describe('Actions', () => {

  it('renders <div> containing <span> for each action', () => {
    const tree = renderer.create(
      <Actions />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onAction when people click each <span>', () => {
    const onAction = jest.fn();
    const component = TestUtils.renderIntoDocument(
      <Wrap><Actions onAction={onAction} /></Wrap>
    );

    const spans = TestUtils.scryRenderedDOMComponentsWithTag(component, 'span')
    spans.forEach(span => TestUtils.Simulate.click(span));

    const calls = onAction.mock.calls;
    expect(calls.length).toEqual(spans.length);
    expect(calls[0][0]).toEqual('display');
    expect(calls[1][0]).toEqual('update');
    expect(calls[2][0]).toEqual('delete');
  });

});
