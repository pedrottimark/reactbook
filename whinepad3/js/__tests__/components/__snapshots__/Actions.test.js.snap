exports[`Actions renders <div> containing <span> for each action 1`] = `
<div
  className="Actions">
  <span
    className="ActionsInfo"
    onClick={[Function]}
    tabIndex="0"
    title="More info">
    ℹ
  </span>
  <span
    className="ActionsEdit"
    onClick={[Function]}
    tabIndex="0"
    title="Edit">
    ✐
  </span>
  <span
    className="ActionsDelete"
    onClick={[Function]}
    tabIndex="0"
    title="Delete">
    x
  </span>
</div>
`;
